src/core/util/byte_stream.cc

```diff
@@ -100,6 +101,11 @@ void OutputByteStream::ProduceData(std::size_t amount) {
 }
 
 void OutputByteStream::WriteData(const std::uint8_t* data, std::size_t len) {
+  if (!data || !len)
+    {
+      LOG(debug) << "OutputByteStream: skip null data or size";
+      return;
+    }
```

Comment:
>BOOST_CHECK_NO_THROW(output.WriteData(nullptr, 1));

As mentioned here https://github.com/monero-project/kovri/pull/672#discussion_r127573419, I'm having a real hard time *not* justifying we throw. Example:

```cpp
std::array<std::uint8_t, 1> arr {{}};
OutputByteStream obs(array.data(), array.size());

auto data_ptr = pointer_that_we_expect_to_NEVER_be_nullptr_but_for_some_extraneous_reason_is_unwittingly_nullptr;
obs.WriteData(data_ptr, 1);  // Doesn't throw - but we really should
```

Outside of the use-case example you proposed in the above issue comment, when would we need to *purposefully* write any non-zero length of a nullptr (it sounds crazy!). And now we'll need to do checks in *all* caller functions? I *completely* understand what you want to do with SSU but, like I mentioned in the above comment, maybe that should be done SSU-side and not with byte stream.

Am I misunderstanding your intention? If not, can we devise a less questionable solution?